### PR TITLE
Np 9071 update cristin api client

### PR DESCRIPTION
**Pull request by svenngjeto**
Update of parts of project in Cristin, filelds updated are title, contributors, coordinating_institution, language, start_date, end_date

### CODE REVIEW

**Review by torarnet — DISMISSED** [review]

File: cristin-project/src/main/java/no/unit/nva/cristin/projects/CristinProjectPatchJsonCreator.java [hunk 1]
```diff
@@ -1,55 +1,105 @@
...
+            output.set(TITLE, OBJECT_MAPPER.valueToTree(titles));
+        }
+    }
+
+    private void addCoordinatingInstitutionIfPresent() {
+        if (input.has(COORDINATING_INSTITUTION)) {
+            final String content = input.get(COORDINATING_INSTITUTION).asText();
+            Organization coordinatingInstitution =
+                    attempt(() -> OBJECT_MAPPER.readValue(content, Organization.class)).orElseThrow();
+            CristinOrganization cristinOrganization = fromOrganizationContainingInstitution(coordinatingInstitution);
```
Comment (brinxmat) [inline cristin-project/src/main/java/no/unit/nva/cristin/projects/CristinProjectPatchJsonCreator.java:70]: Don't mix types and var, use var

File: cristin-project/src/main/java/no/unit/nva/cristin/projects/CristinProjectPatchJsonCreator.java [hunk 2]
```diff
@@ -1,55 +1,105 @@
...
+        }
+    }
+
+    private void addContributorsIfPresent() {
+        if (input.has(CONTRIBUTORS)) {
+            TypeReference<List<NvaContributor>> typeRef = new TypeReference<>() {
+            };
+            List<NvaContributor> contributors =
+                    attempt(() -> OBJECT_MAPPER.readValue(input.get(CONTRIBUTORS).asText(), typeRef)).orElseThrow();
+            List<CristinPerson> participants = extractContributors(contributors);
```
Comment (brinxmat) [inline cristin-project/src/main/java/no/unit/nva/cristin/projects/CristinProjectPatchJsonCreator.java:81]: Var

File: cristin-project/src/main/java/no/unit/nva/cristin/projects/UpdateCristinApiClient.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    /**
+     * perform an PATCH operation for a Cristin project.
+     * @param projectId identifier for project in Cristin
+     * @param cristinJson structure holding patch data
+     * @return Void
+     * @throws ApiGatewayException when operation fails
+     */
+    public Void updateProjectInCristin(String projectId, ObjectNode cristinJson) throws ApiGatewayException {
+        URI uri = generateCristinUri(projectId);
+        HttpResponse<String> response = patch(uri, cristinJson.toString());
```
Comment (brinxmat) [inline cristin-project/src/main/java/no/unit/nva/cristin/projects/UpdateCristinApiClient.java:45]: var

File: cristin-project/src/test/java/no/unit/nva/cristin/projects/UpdateCristinProjectHandlerTest.java [hunk 4]
```diff
@@ -124,6 +129,22 @@ void shouldReturnBadRequestWhenNoSupportedFieldsArePresent() throws IOException
...
+    void shouldReturnBadRequestWhenTitleHasNoLanguageFieldPresent() throws IOException {
+        ObjectNode jsonObject = OBJECT_MAPPER.createObjectNode();
+        jsonObject.put(TITLE, randomString());
+        GatewayResponse<Void> gatewayResponse = sendQuery(validPath, jsonObject.toString());
+        assertEquals(HTTP_BAD_REQUEST, gatewayResponse.getStatusCode());
+    }
+
+    @Test
+    void shouldReturnOKNoContentWhenRequestHasTitleAndLanguagePresent() throws IOException {
+        ObjectNode jsonObject = OBJECT_MAPPER.createObjectNode();
```
Comment (brinxmat) [inline cristin-project/src/test/java/no/unit/nva/cristin/projects/UpdateCristinProjectHandlerTest.java:142]: Var

**Review by brinxmat — CHANGES_REQUESTED** [review]

**Review by brinxmat — APPROVED** [review]

**Pull request merged**